lib/MusicBrainz/Server/Data/Artist.pm

```diff
@@ -101,14 +101,21 @@ sub find_by_subscribed_editor
 
 sub find_by_area {
     my ($self, $area_id, $limit, $offset) = @_;
+    my (
+        $containment_query,
+        @containment_query_args,
+    ) = $self->c->model('Area')->get_containment_query('$2', 'any(array[area, begin_area, end_area])');
     my $query = "SELECT " . $self->_columns . "
                  FROM " . $self->_table . "
-                    LEFT JOIN area ON artist.area = area.id
-                    LEFT JOIN area begin_area ON artist.begin_area = begin_area.id
-                    LEFT JOIN area end_area ON artist.end_area = end_area.id
```

Comment:
All these joins seem completely unnecessary - the query does nothing with the area tables after this.